File: app/src/main/java/com/nimtego/itunes/presentation/main/songs/SongTabsFragment.java
package com.nimtego.itunes.presentation.main.songs;

import android.content.Context;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import com.nimtego.itunes.presentation.main.fragments.MainTabsFragment;
import com.nimtego.itunes.presentation.main.model.SongModel;

import java.util.ArrayList;
import java.util.Collection;

public class SongTabsFragment
        extends MainTabsFragment<SongContract.Presenter>
        implements SongContract.View<SongContract.Presenter>  {

    @Override
    protected RecyclerView.LayoutManager rvLayoutManager(Context context) {
        return new LinearLayoutManager(context);
    }

    @Override
    public void render(Collection<SongModel> songModels) {
        RecyclerView.Adapter adapter = new SongAdapter(new ArrayList<>(songModels),
                this.getActivity());
        mRecyclerView.setAdapter(adapter);
    }

    @Override
    public void search(String response) {
        mPresenter.search(response);
    }

    @Override
    public SongContract.Presenter supplyPresenter() {
        return new SongPresenter();
    }
}

File: app/src/main/java/com/nimtego/plectrum/presentation/old/main/songs/SongTabsFragment.kt
package com.nimtego.plectrum.presentation.old.main.songs

import android.content.Context
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import com.arellomobile.mvp.presenter.InjectPresenter
import com.arellomobile.mvp.presenter.ProvidePresenter
import com.nimtego.plectrum.presentation.old.main.fragments.MainTabsFragment
import com.nimtego.plectrum.presentation.old.main.model.SongModel
import com.nimtego.plectrum.presentation.ui.widget.SpaceItemDecorator

class SongTabsFragment : MainTabsFragment(), SongTabView {

    @InjectPresenter
    internal lateinit var presenter: SongPresenterImpl

    @ProvidePresenter
    fun provideRepositoryPresenter(): SongPresenterImpl {
        return SongPresenterImpl()
    }

    override fun render(songModels: Collection<SongModel>) {
        val adapter = SongAdapter(ArrayList(songModels),
                this.activity)
        adapter.setOnItemClickListener( object : SongAdapter.OnItemClickListener {
            override fun onUserItemClicked(songModel: SongModel) {
                presenter.itemClick(songModel)
            }
        })
        mRecyclerView?.adapter = adapter
    }

    override fun itemDecorator(): RecyclerView.ItemDecoration {
        return SpaceItemDecorator(1,
                20,
                true)
    }

    override fun rvLayoutManager(context: Context): RecyclerView.LayoutManager {
        return LinearLayoutManager(context)
    }

    override fun search(response: String) {
        presenter.search(response)
    }

    companion object {

        fun getInstance(response: String): SongTabsFragment {
            val fragment =SongTabsFragment()
            val arguments = Bundle()
            arguments.putString(RESPONSE, response)
            fragment.arguments = arguments
            return fragment
        }
    }
}